### PR TITLE
Updating Roslyn deps

### PR DESCRIPTION
**Pull request by psfinaki**
This should include [this](https://github.com/dotnet/roslyn/pull/66926) and [this](https://github.com/dotnet/roslyn/pull/66874). Validated this locally according to the DEVGUIDE.

Also removed several random version props for packages we don't use anywhere.

### CODE REVIEW

**Comment by vzarytovskii** [comment]
Yaml is not yamling: https://dev.azure.com/dnceng-public/public/_build/results?buildId=186427&view=results

**Comment by psfinaki** [comment]
Damn, I hate YAML. Thanks

File: azure-pipelines.yml [hunk 1]
```diff
@@ -346,9 +346,9 @@ stages:
               vs_release:
                 _configuration: Release
                 _testKind: testVs
-              inttests_release:
-                _configuration: Release
-                _testKind: testIntegration
+              # inttests_release:
```
Comment (KevinRansom) [inline azure-pipelines.yml:349]: Can you just remove these rather than comment them out.

**Review by KevinRansom — APPROVED** [review]
Apart from removing the commented-out piece this is fine.

**Review by 0101 — APPROVED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (psfinaki) [reply azure-pipelines.yml]: There you go

**Pull request merged**